tests/providers/google/cloud/operators/test_compute.py

```diff
@@ -864,10 +1458,208 @@ def test_missing_name(self, mock_hook):
     "minReadySec": 1800,
 }
 
+GCE_INSTANCE_GROUP_MANAGER_BODY_API_CALL = {
+    "name": GCE_INSTANCE_GROUP_MANAGER_NAME,
+    "base_instance_name": GCE_INSTANCE_GROUP_MANAGER_NAME,
+    "instance_template": GCE_INSTANCE_TEMPLATE_SOURCE_URL,
+    "target_size": 1,
+}
+
+
+class TestGceInstanceGroupManagerInsert:
+    @mock.patch(COMPUTE_ENGINE_HOOK_PATH)
+    def test_insert_igm_should_execute_successfully(self, mock_hook):
+        get_instance_group_manager_obj_mock = mock.MagicMock()
+        get_instance_group_manager_obj_mock.__class__ = InstanceGroupManager
+        mock_hook.return_value.get_instance_group_manager.side_effect = [
+            NotFound("Error message"),
+            get_instance_group_manager_obj_mock,
+        ]
+        op = ComputeEngineInsertInstanceGroupManagerOperator(
+            project_id=GCP_PROJECT_ID,
+            body=GCE_INSTANCE_GROUP_MANAGER_BODY_API_CALL,
+            zone=GCE_ZONE,
+            task_id=TASK_ID,
+            retry=RETRY,
+            timeout=TIMEOUT,
+            metadata=METADATA,
+            gcp_conn_id=GCP_CONN_ID,
+            impersonation_chain=IMPERSONATION_CHAIN,
+        )
+        op.execute(context=mock.MagicMock())
+        mock_hook.assert_called_once_with(
+            api_version=API_VERSION,
+            gcp_conn_id=GCP_CONN_ID,
+            impersonation_chain=IMPERSONATION_CHAIN,
+        )
+        mock_hook.return_value.insert_instance_group_manager.assert_called_once_with(
+            zone=GCE_ZONE,
+            project_id=GCP_PROJECT_ID,
+            body=GCE_INSTANCE_GROUP_MANAGER_BODY_API_CALL,
+            request_id=None,
+        )
+
+    @mock.patch(COMPUTE_ENGINE_HOOK_PATH)
+    def test_insert_igm_should_throw_ex_when_missing_project_id(self, mock_hook):
+        with pytest.raises(AirflowException) as ctx:
+            op = ComputeEngineInsertInstanceGroupManagerOperator(
+                project_id="",
+                body=GCE_INSTANCE_GROUP_MANAGER_BODY_API_CALL,
+                zone=GCE_ZONE,
+                task_id=TASK_ID,
+                retry=RETRY,
+                timeout=TIMEOUT,
+                gcp_conn_id=GCP_CONN_ID,
+                impersonation_chain=IMPERSONATION_CHAIN,
+            )
+            op.execute(context=mock.MagicMock())
+        err = ctx.value
+        assert "The required parameter 'project_id' is missing" in str(err)
```

Comment:
You can use match parameter to pytest.raises to catch exceptions with error message, instead of doing separate assertion

Reply:
Done